### PR TITLE
[enhancement](chore) build fe-common when build java-udf

### PR DESCRIPTION
**Pull request by morrySnow**
# Proposed changes

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -451,7 +452,11 @@ if [[ "${FE_MODULES}" != '' ]]; then
     if [[ "${CLEAN}" -eq 1 ]]; then
         clean_fe
     fi
-    "${MVN_CMD}" package -pl ${FE_MODULES:+${FE_MODULES}} -DskipTests
+    MVN_PARAMS="-DskipTests"
+    if [[ "$(uname -s)" = 'Darwin' ]]; then
+      MVN_PARAMS="${MVN_PARAMS} -Dos.arch=x86_64"
+    fi
```
Comment (adonis0147) [inline build.sh:458]: It seems that we don't need this option `-Dos.arch=x86_64` anymore. See #13571.

**Comment by morrySnow** [comment]
already fixed by #13571

**Review by adonis0147 — APPROVED** [review]
LGTM

File: build.sh [hunk 2]
```diff
@@ -350,6 +350,7 @@ if [[ "${BUILD_SPARK_DPP}" -eq 1 ]]; then
     modules+=("spark-dpp")
 fi
 if [[ "${BUILD_JAVA_UDF}" -eq 1 ]]; then
+    modules+=("fe-common")
```
Comment (adonis0147) [inline build.sh:353]:
```suggestion
    if [[ "${BUILD_FE}" -eq 0 ]]; then
        modules+=("fe-common")
        modules+=("fe-core")
    fi
```

**Review by adonis0147 — APPROVED** [review]
LGTM

**Pull request merged**